posts/2023-10-20-quarkus-benelux-rotterdam-oct-2023.adoc

```diff
@@ -0,0 +1,82 @@
+= Quarkus Benelux in Rotterdam
+LeonorLunatech
+v1.0, 2023-10-20
+:title: Quarkus Benelux in Rotterdam
+:imagedir: ../media/2023-10-13-quarkus-benelux-rotterdam-oct-2023
+:lang: en
+:tags: [quarkus, java, web-applications]
+
+=== Introduction
+Last Thursday, the https://www.meetup.com/quarkus-benelux/[Quarkus Benelux] meetup talks were hosted at our Rotterdam office.
+Now that the Covid-19 lockdowns are a thing of the past,
+I noticed this to be a more dynamic event, the attendance was higher and there was more energy in the air, in general.
+I think the tasty drinks and the warm food added to the fun too.
+
+There were two very interesting talks: `Quarkus on Java 21` by https://twitter.com/wjglerum[Willem Jan Glerum] and
+`The Quarkus Rainbow for FullStack/Web development` by https://twitter.com/Ia3andy[Andy Damevin].
+
+=== Quarkus on Java 21 by Willem Jan Glerum
+Willem Jan, a software engineer at Lunatech, gave an insightful and informative presentation.
+He began by familiarizing the audience with Java’s evolution in
+terms of release cadence and the varying maturity levels of the features available in each Java release.
+
+For the rest of the talk, Willem focused on the fully delivered features from https://ambr.top/en[Project Amber] and https://openjdk.org/projects/loom/[Project Loom]:
+
+- https://openjdk.org/jeps/440[Record Patterns]
+- https://openjdk.org/jeps/441[Pattern Matching for switch]
+- https://openjdk.org/jeps/444[Virtual Threads]
+
+We examined both simple and complex examples of how `Record Patterns` and
+`Pattern Matching for switch` can be utilized. Willem guided us through the different _flavours_ of threads like `OS thread`, `Platform thread`,
+`Virtual Thread`, `Carrier Thread`. It may sound like a lot of information, but Willem’s clear diagrams made the talk easy to follow.
+
+We also explored how `Quarkus` enables the writing of concurrent applications in a blocking, reactive, or virtual threads manner.
+Switching between these styles of concurrent application development is as simple as using annotations!
+However, these annotations should not be used indiscriminately. As the saying goes, ‘With great power comes great responsibility.’
+Both reactive and virtual threads coding styles have their strengths and weaknesses, and should be chosen carefully, based on the specific use case.
```

Comment:
I typically try to use the conjunction "and" only once in a sentence to avoid any chances of it even seeming like a run-on sentence. I would recommend something like: "Both reactive and virtual threads coding styles have their strengths as well as weaknesses, which is why they should be chosen carefully based on the specific use case."

Reply:
Thanks! I like it more as well